src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md

```diff
@@ -8,45 +8,53 @@ This is a literate Rzk file:
 
 ## Problem statement
 
-Derive the recursion principle for products
-$\mathsf{rec}_{A \times B}$ using only the projections,
-and verify that the definitional equalities are valid.
-Do the same for $\Sigma$-types.
+!!! question "Exercise 1.2"
+    Derive the recursion principle for products
+    $\mathsf{rec}_{A \times B}$ using only the projections,
+    and verify that the definitional equalities are valid.
+    Do the same for $\Sigma$-types.
 
 ## Solution for products
 
-In this solution, we use `prod`, defined earlier in [chapter 1.5](../05-product-types.rzk.md).
+In this solution, we use `prod`, defined earlier in [section 1.5](../05-product-types.rzk.md#define:prod).
 
 Introducing constructor for values of `prod` type:
+
 ```rzk
 #def pair
   (A B : U)
   : A -> B -> prod A B
   := \ a b -> (a, b)
 ```
 
-Projections were also defined in [chapter 1.5](../05-product-types.rzk.md) (`pr₁`, `pr₂`).
-
-Type of the recursor on $A \times B$ is 
+The type of the recursor on $A \times B$ is
 
 $$
-\mathsf{rec}_{A \times B}: \prod_{C:\mathcal{U}} (A \rightarrow B \rightarrow C) \rightarrow A \times B \rightarrow C
+\mathsf{rec}_{A \times B}: \prod_{C:\mathcal{U}} (A \to B \to C) \to A \times B \to C
 $$
 
-Defining it through projections:
+It can be defined using projection as such:
+
+$$ rec_{A\times B}(C, g, x) :\equiv g(\mathsf{pr}_1(x))(\mathsf{pr}_2(x)) $$
+
+And represented in rzk like so:
+
 ```rzk
 #def prod-rec-via-projections
   (A B : U)
   (C : U)
-  : (g : A -> B -> C) -> (p : prod A B) -> C
-  := \ g p -> g (pr₁-via-rec A B p) (pr₂-via-rec A B p)
+  (g : A -> B -> C)
+  (p : prod A B)
+  : C
+  := g (pr₁-via-rec A B p) (pr₂-via-rec A B p)
```

Comment:
oh, I mixed up the definitions; `pr₁` and `pr₂` should've been used here instead. Can you add it here in order not to forget it later?)
```suggestion
  := g (pr₁ A B p) (pr₂ A B p)
```

Reply:
and since we're on it, return back https://github.com/rzk-lang/hottbook/pull/8/files#diff-7daa1f51cc3a6951783fac34721d4e07dee355b9b8b79782026da7faa3494307L28
much appreciated :)

Reply:
Yeah, I was going to ask about that)
But is there a reason to use $pr_1$ instead of the built-in `first`?

Reply:
`first` is not a function, and is a (part of) builtin primitive syntax (subject to change). So yes, probably it should not be used except to define projection functions :)